src/plugins/yawn/token.hpp

```diff
@@ -0,0 +1,109 @@
+/**
+ * @file
+ *
+ * @brief This file contains the declaration of a class that stores data
+ *        about a token produced by a lexer.
+ *
+ * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ */
+
+#ifndef ELEKTRA_PLUGIN_YAWN_TOKEN_HPP
+#define ELEKTRA_PLUGIN_YAWN_TOKEN_HPP
+
+// -- Imports ------------------------------------------------------------------------------------------------------------------------------
+
+#include <string>
+
+#include "location.hpp"
+#include "token.hpp"
+
+// -- Class --------------------------------------------------------------------------------------------------------------------------------
+
+namespace yawn
+{
+
+/** This class represents a token emitted by a lexer. */
+class Token
+{
+
+	/** This variable stores the location of this token in the scanned text. */
+	Location _location;
```

Comment:
The C++ standard disallows identifier to start with _. If you used this everywhere now, you can keep it.

Reply:
> The C++ standard disallows identifier to start with _.

Isn’t that [only true for the global namespace](https://stackoverflow.com/questions/228783/what-are-the-rules-about-using-an-underscore-in-a-c-identifier)?

Reply:
If you only start with lowercase you should be fine.